addons/test_mail/tests/test_message_track.py

```diff
@@ -283,6 +289,39 @@ def test_tracked_compute(self):
             ('partner_email', 'char', 'foo@example.com', 'bar@example.com'),
         ])
 
+    def test_message_track_monetary(self):
+        """ Update a record with a tracked monetary field """
+        # Create companies with different currencies
+        new_company = self.env['res.company'].create({
+            'name': 'new company',
+            'currency_id': 2,
+        })
+
+        # Check if the tracking value have the correct currency and values
+        self.record.write({
+            'revenue': 100,
+        })
+        self.flush_tracking()
+        self.assertEqual(len(self.record.message_ids), 1)
+        # The currency didn't change but the value did
+        self.assertEqual(self.record.message_ids.tracking_value_ids.currency_id.id, 1)
+        self.assertEqual(self.record.message_ids.tracking_value_ids.currency_id.id, 1)
+        self.assertEqual(self.record.message_ids.tracking_value_ids.old_value_monetary, 0)
+        self.assertEqual(self.record.message_ids.tracking_value_ids.new_value_monetary, 100)
+
+        # Check if the tracking value have the correct currency and values after changing the value and the company
+        self.record.write({
+            'revenue': 200,
+            'company_id': new_company.id,
+        })
+        self.flush_tracking()
+        self.assertEqual(len(self.record.message_ids), 2)
+        tracking_values = self.record.message_ids[0].tracking_value_ids.filtered(lambda values: values.new_value_monetary)
```

Comment:
Don't you have a nicer way of filtering?
Like "field_type" == "monetary" maybe? Or better "field.name" == "revenue"?